src/NHibernate/Properties/FieldAccessor.cs

```diff
@@ -331,6 +331,11 @@ public void Emit(ILGenerator il)
 			{
 				il.Emit(OpCodes.Stfld, field);
 			}
+
+			internal bool CanEmit()
+			{
+				return !field.IsInitOnly;
```

Comment:
Is this actually supported only for id/parent properties? Otherwise I do not understand how it works for the optimized `setters` we currently already have.

Reply:
My mistake, I didn't use the owner argument when creating a `DynamicMethod` and without the owner you can't set a readonly field. I forgot to use the `CreateDynamicMethod` in `ReflectionOptimizer` when copy/pasting the code from the previous PR.